src/unique_arc.rs

```diff
@@ -63,6 +58,42 @@ impl<T> UniqueArc<T> {
             })
         }
     }
+
+    /// Gets the inner value of the unique arc
+    pub fn into_inner(this: Self) -> T {
+        let result =
```

Comment:
this should be inside the debug assert, otherwise it will always get run

Reply:
we should also `mem::forget(this)`

Reply:
It has to always run because of synchronization, the `Acquire` success load ensures that all other threads have released their references to the arc. The forgetting isn't needed since `Arc`'s drop routine only triggers if drop is called while the ref count is equal to one, and the compare swap sets the ref count to zero since we're essentially doing a manual drop here

Reply:
It's a `UniqueArc`, the only way you can obtain one is _if_ the other threads have released their references.